pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -276,8 +276,9 @@ func createClusterAccessSecret(clientset kubernetes.Interface, namespace string,
 	}
 	secret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      AppName,
```

Comment:
In addition to adding the finalizer, we will also need:
1. Instructions to remove the finalizers in the cleanup guide: ceph-teardown.md
2. The integration tests need to remove the finalizers or else they will likely wait forever and timeout
3. If the cephcluster is [cleaning up the cluster](https://github.com/rook/rook/blob/e4ac38e55538fdb7385cdb177c8edec9679cdaa9/pkg/operator/ceph/cluster/controller.go#L274), we should remove the finalizers so the intentional deletion is not blocked.

Reply:
> 3. If the cephcluster is cleaning up the cluster, we should remove the finalizers so the intentional deletion is not blocked.

Added the delete option, 
How can I test this, how this is called?

Reply:
@parth-gr delete the ceph cluster and see if deletions is stuck

Reply:
Thank @subhamkrai, it's been deleted successfully.